pkg/components/metallb/component.go

```diff
@@ -55,88 +55,42 @@ func (c *component) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContex
 	if configBody == nil {
 		return hcl.Diagnostics{}
 	}
+
 	return gohcl.DecodeBody(*configBody, evalContext, c)
 }
 
-// TODO: Convert to Helm chart.
 func (c *component) RenderManifests() (map[string]string, error) {
-	// Here are `nodeSelectors` and `tolerations` that are set by upstream. To make sure that we
-	// don't miss them out we set them manually here. We cannot make these changes in the template
-	// because we have parameterized these fields.
-	if c.SpeakerNodeSelectors == nil {
-		c.SpeakerNodeSelectors = map[string]string{}
-	}
-	// MetalLB only supports Linux, so force this selector, even if it's already specified by the
-	// user.
-	c.SpeakerNodeSelectors["beta.kubernetes.io/os"] = "linux"
-
-	if c.ControllerNodeSelectors == nil {
-		c.ControllerNodeSelectors = map[string]string{}
+	helmChart, err := components.Chart(Name)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving chart from assets: %w", err)
 	}
-	c.ControllerNodeSelectors["beta.kubernetes.io/os"] = "linux"
-	c.ControllerNodeSelectors["node.kubernetes.io/master"] = ""
-
-	c.ControllerTolerations = append(c.ControllerTolerations, util.Toleration{
-		Effect: "NoSchedule",
-		Key:    "node-role.kubernetes.io/master",
-	})
 
 	t, err := util.RenderTolerations(c.SpeakerTolerations)
 	if err != nil {
 		return nil, fmt.Errorf("marshaling speaker tolerations: %w", err)
 	}
+
 	c.SpeakerTolerationsJSON = t
 
 	t, err = util.RenderTolerations(c.ControllerTolerations)
 	if err != nil {
 		return nil, fmt.Errorf("rendering controller tolerations: %w", err)
 	}
-	c.ControllerTolerationsJSON = t
 
-	controllerStr, err := template.Render(deploymentController, c)
-	if err != nil {
-		return nil, fmt.Errorf("rendering controller Deployment template: %w", err)
-	}
+	c.ControllerTolerationsJSON = t
 
-	speakerStr, err := template.Render(daemonsetSpeaker, c)
+	values, err := internaltemplate.Render(chartValuesTmpl, c)
 	if err != nil {
-		return nil, fmt.Errorf("rendering speaker DaemonSet template: %w", err)
+		return nil, fmt.Errorf("rendering values template failed: %w", err)
```

Comment:
See https://github.com/kinvolk/lokomotive/issues/59#issuecomment-648894086.
```suggestion
		return nil, fmt.Errorf("rendering values template: %w", err)
```